### PR TITLE
✨ execute_deposit_utils functions

### PR DESCRIPTION
**Pull request by kasteph**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

## What is the current behavior?

Resolves: #354

## What is the new behavior?

- Implement the `execute_deposit` function in the `execute_deposit_utils` library.
- Implement the `_execute_deposit` function in the `execute_deposit_utils` library.
- Implement the `swap` function in the `execute_deposit_utils` library.

## Does this introduce a breaking change?

No

## Other information

Still working on adding some basic unit test

### CODE REVIEW

File: src/deposit/execute_deposit_utils.cairo [hunk 1]
```diff
@@ -80,29 +104,390 @@ struct ExecuteDepositCache {
...
+        ui_deposit_fee_type(),
+    );
+
+    (*params.event_emitter)
+        .emit_swap_fees_collected(
+            *_params.market.market_token,
+            *_params.token_in,
+            *_params.token_in_price.min,
+            'deposit',
+            fees,
```
Comment (kasteph) [inline src/deposit/execute_deposit_utils.cairo:308]: @sparqet when building the project, I'm getting the following error here:

```
error: Variable was previously moved.
 --> execute_deposit_utils.cairo:308:13
            fees
            ^**^
note: variable was previously used here:
  --> execute_deposit_utils.cairo:308:13
            fees
            ^**^
```

I'm a bit baffled by this as this and the lines before are practically the same as [`swap_utils#L208-L233`](
https://github.com/keep-starknet-strange/satoru/blob/main/src/swap/swap_utils.cairo#L208-L233)
and [`swap_utils#L351-L365`](https://github.com/keep-starknet-strange/satoru/blob/main/src/swap/swap_utils.cairo#L351-L365). Do you have any idea where the ownership/moving error might come from?

File: src/deposit/execute_deposit_utils.cairo [hunk 2]
```diff
@@ -80,29 +104,390 @@ struct ExecuteDepositCache {
...
 /// # Arguments
-/// * `params` - ExecuteDepositParams.
-/// * `_params` - _ExecuteDepositParams.
+/// * `params` - @ExecuteDepositParams.
+/// * `_params` - @_ExecuteDepositParams.
 #[inline(always)]
-fn _execute_deposit(params: ExecuteDepositParams, _params: _ExecuteDepositParams) -> u128 {
-    //TODO
-    0
+fn _execute_deposit(params: @ExecuteDepositParams, _params: @_ExecuteDepositParams) -> u128 {
```
Comment (Sk8erboi84) [inline src/deposit/execute_deposit_utils.cairo:274]: Change the name of the function please we decided not to use the _ at the beginning of the function name

File: src/deposit/execute_deposit_utils.cairo (hunk 1 above)
Comment (sparqet) [reply src/deposit/execute_deposit_utils.cairo:308]: You need to add Copy in the derive of SwapFees struct

File: src/deposit/execute_deposit_utils.cairo (hunk 2 above)
Comment (kasteph) [reply src/deposit/execute_deposit_utils.cairo:274]: @Sk8erboi84 sounds good, I didn't change anything in the function signature and just worked from there (it's also the same in gmx). What should I change it to?

Comment (Sk8erboi84) [reply src/deposit/execute_deposit_utils.cairo:274]: Change it to a coherent function name

**Comment by sparqet** [comment]
Hi @kasteph ! How is it going on ?

**Comment by kasteph** [comment]
@sparqet it's ready

**Comment by sparqet** [comment]
> @sparqet ~it's ready~ nvm, it's failing build again 😭

It is just the coding style

File: src/deposit/execute_deposit_utils.cairo [hunk 3]
```diff
@@ -80,29 +104,392 @@ struct ExecuteDepositCache {
...
+        }
+    }
+
+    mint_amount +=
+        market_utils::usd_to_market_token_amount(
+            amount_after_fees * *_params.token_in_price.min,
+            to_unsigned(pool_value),
+            market_tokens_supply,
+        );
+
```
Comment (sparqet) [inline src/deposit/execute_deposit_utils.cairo:453]: Missing call to MarketUtils.applyDeltaToPoolAmount and MarketUtils.validatePoolAmount

File: src/event/event_utils.cairo [hunk 4]
```diff
@@ -5,10 +5,10 @@ use traits::Default;
 use satoru::utils::traits::ContractAddressDefault;
 
 //TODO Switch the append with a set in the functions when its available
-
-#[derive(Default, Drop, Serde)]
+#[derive(Drop, Serde)]
 struct EventLogData {
     cant_be_empty: u128, // remove 
+    uint_items: UintItems,
```
Comment (sparqet) [inline src/event/event_utils.cairo:11]: remove uint_items and use LogData struct

**Comment by kasteph** [comment]
@sparqet it's ready again 🙏🏽

**Review by sparqet — APPROVED** [review]

**Pull request merged**